xtra-libp2p/src/endpoint.rs

```diff
@@ -492,7 +529,55 @@ impl Endpoint {
     async fn handle(&mut self, msg: NewListenAddress) {
         // FIXME: This address could be a "catch-all" like "0.0.0.0" which actually results in
         // listening on multiple interfaces.
-        self.listen_addresses.insert(msg.listen_address);
+        self.listen_addresses.insert(msg.listen_address.clone());
+        self.notify_listen_address_added(msg.listen_address).await;
+    }
+}
+
+impl Endpoint {
+    async fn notify_connection_established(&mut self, peer: PeerId) {
+        for subscriber in &self.subscribers.connection_established {
+            if let Err(e) = subscriber
+                .send_async_safe(ConnectionEstablished { peer })
+                .await
+            {
+                tracing::error!("Unable to reach subscriber: {e:#}");
```

Comment:
fair enough!

Reply:
True in the context of this actor, but knowing the architecture this should not happen. We deliberately expect the subscribers always to be available and don't actively remove them, because it's expected that they are supervised. I would keep this as error because this should not happen.